test/e2e/e2e_test.go

```diff
@@ -0,0 +1,147 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/xmudrii/etcdproxy-controller/pkg/apis/etcd/v1alpha1"
+	clientset "github.com/xmudrii/etcdproxy-controller/pkg/client/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// kubeconfigPath is path to the kubernetes configuration file.
+var kubeconfigPath = filepath.Join(os.Getenv("HOME"), "/.kube/config")
```

Comment:
Is there a better way for providing `kubeconfigPath`? Maybe with an environment variable, or this is good enough?

Reply:
> Is there a better way for providing kubeconfigPath? Maybe with an environment variable, or this is good enough?

We respect `KUBECONFIG` in most (all?) kube binaries.  I'd use that.

Reply:
This is a great idea, updated the PR to use it.